### PR TITLE
Add vegan_rarefaction tool.

### PR DESCRIPTION
**Pull request by blankenberg**
Needs tests and more comprehensive help.

May need to be modified based upon the R package that is accepted by IUC.

### CODE REVIEW

**Comment by davebx** [comment]
:+1:

File: tools/vegan_rarefaction/vegan_rarefaction.xml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        <data format="tabular" name="output_species_frequency" label="${tool.name} on ${on_string} (frequency of species)"/>
+        <data format="tabular" name="output_richness" label="${tool.name} on ${on_string} (estimated richness)"/>
+        <data format="tabular" name="output_species_probabilities" label="${tool.name} on ${on_string} (species probabilities)"/>
+        <data format="tabular" name="output_slope" label="${tool.name} on ${on_string} (slope of curve)"/>
+        <data format="txt" name="vegan_rarefaction_script_output" label="${tool.name} on ${on_string} (Rscript)" />
+        <data format="png" name="output_plot" label="${tool.name} on ${on_string} (plot)" />
+    </outputs>
+    <tests>
+        <test>
+            <param name="fastq_input|fastq_input_selector" value="single" />
```
Comment (erasche) [inline tools/vegan_rarefaction/vegan_rarefaction.xml]: How does this work? I'm not seeing "fastq_input" in this file, nor a macro being loaded.

Comment (davebx) [inline tools/vegan_rarefaction/vegan_rarefaction.xml]: It appears as though this may be copy/pasted from @blankenberg's cd-hit-dup wrapper.

**Comment by nekrut** [comment]
:+1:

**Comment by bgruening** [comment]
@blankenberg is this still WIP?

**Comment by blankenberg** [comment]
removed WIP tagging

File: tools/vegan_rarefaction/tool_dependencies.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0"?>
+<tool_dependency>
+  <package name="vegan" version="2.3-0">
```
Comment (bgruening) [inline tools/vegan_rarefaction/tool_dependencies.xml]: Indentation.

File: tools/vegan_rarefaction/vegan_rarefaction.xml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    <requirements>
+        <requirement type="package" version="3.2.1">R</requirement>
+        <requirement type="package" version="2.3-0">vegan</requirement>
+    </requirements>
+    <stdio>
+        <exit_code range="1:" />
+        <exit_code range=":-1" />
+    </stdio>
+
+    <command>cp "${vegan_rarefaction_script}" "${vegan_rarefaction_script_output}" &amp;&amp;
```
Comment (bgruening) [inline tools/vegan_rarefaction/vegan_rarefaction.xml]: CDATA?

File: tools/vegan_rarefaction/vegan_rarefaction.xml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        <param name="input_abundance" type="data" format="tabular" label="File with abundance values for community" help="Rows are samples; columns are species/phyla/community classifier"/>
+        
+        <!-- spec column label?, counts (spec in rows) single select
+        treatment columns multi select, 1+
+        -->
+        
+        <param name="species_column" label="Group name column" type="data_column" data_ref="input_abundance" value="6" help="Species, phylum, etc"/>
+        <param name="sample_columns" label="Sample count columns" type="data_column" multiple="True" value="2" data_ref="input_abundance" help="Select each column that contains counts"/>
+        <!-- <param name="swap_columns_rows" type="boolean" truevalue="true" falsevalue="false" checked="False" label="Swap rows and columns" help="Rows are species/phyla/community classifier; columns are samples"/> -->
+        <param name="sample_size" type="integer" value="" min="1" optional="True" label="Subsample size for rarefying community" help="leave empty to subsample by community size for the smallest sample"/>
```
Comment (bgruening) [inline tools/vegan_rarefaction/vegan_rarefaction.xml]: s/leave/Leave/ and maybe as sentence.

File: tools/vegan_rarefaction/vegan_rarefaction.xml [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        <!-- <param name="estimate_standard_error" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="False" label="Estimate standard errors"/> -->
+        <!-- <param name="margin" type="integer" value="1" label="Margin for which the index is computed" /> -->
```
Comment (bgruening) [inline tools/vegan_rarefaction/vegan_rarefaction.xml]: Dan can you please add here a note why this is commented?

File: tools/vegan_rarefaction/vegan_rarefaction.xml [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+        
+        <param name="step_size" type="integer" value="1" min="1" label="Step size for sample sizes"/>
+        <param name="xlab" type="text" value="Sample Size" label="X-axis label"/>
+        <param name="ylab" type="text" value="Species" label="Y-axis label"/>
+        <param name="label" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="True" label="Label rarefaction curves by rownames of X"/>
+
+        
+    </inputs>
+    <outputs>
+        <data format="tabular" name="output_species_count" label="${tool.name} on ${on_string} (number of species)"/>
```
Comment (bgruening) [inline tools/vegan_rarefaction/vegan_rarefaction.xml]: Are all outputs always needed? If not can we add a select box to choose which outputs I want to have?

File: tools/vegan_rarefaction/vegan_rarefaction.xml [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+            <output name="output_species_count" ftype="tabular" file="vegan_output_species_count.tabular" />
+            <output name="output_species_frequency" ftype="tabular" file="vegan_output_species_frequency.tabular" />
+            <output name="output_richness" ftype="tabular" file="vegan_output_richness.tabular" />
+            <output name="output_species_probabilities" ftype="tabular" file="vegan_output_species_probabilities.tabular" />
+            <output name="output_slope" ftype="tabular" file="vegan_output_slope.tabular" />
+        </test>
+    </tests>
+    <help>
+        <![CDATA[
+
```
Comment (bgruening) [inline tools/vegan_rarefaction/vegan_rarefaction.xml]: This should be improved, at least a small description of the tool should be added.
An example would be also nice.

File: tools/vegan_rarefaction/vegan_rarefaction.xml [hunk 8]
```diff
@@ -0,0 +1,114 @@
+<tool id="vegan_rarefaction" name="Vegan Rarefaction" version="0.0.1">
+    <requirements>
```
Comment (bgruening) [inline tools/vegan_rarefaction/vegan_rarefaction.xml]: description is missing .... I'm wondering if planemo complains here?

**Comment by martenson** [comment]
:+1:

**Pull request merged**